shim/shim.c

```diff
@@ -432,6 +446,7 @@ handle_proxy_output(struct cc_shim *shim)
 		code = *(buf + STREAM_HEADER_SIZE); 	// hyperstart has sent the exit status
 		shim_debug("Exit status for container: %d\n", code);
 		free(buf);
+		restoreTerminal();
```

Comment:
Sure. I'm talking about handling signals being delivered to the shim itself (including `SIGSEGV`) - we could make the shim catch these and call `restoreTerminal()`. It might be best to register `restoreTermina()` with `atexit()` too since currently it looks like if the shim fails to read the proxy control line, the shim will exit leaving the terminal in raw mode.